src/main/java/org/gridsuite/study/server/elasticsearch/EquipmentInfosService.java

```diff
@@ -106,6 +114,83 @@ public long getEquipmentInfosCount() {
         return equipmentInfosRepository.count();
     }
 
+    private CompositeAggregation buildCompositeAggregation(String field, Map<String, FieldValue> afterKey) {
+        List<Map<String, CompositeAggregationSource>> sources = List.of(
+                Map.of(field, CompositeAggregationSource.of(s -> s.terms(t -> t.field(field + ".keyword")))
+                )
+        );
+
+        CompositeAggregation.Builder compositeAggregationBuilder = new CompositeAggregation.Builder()
+                .size(AGGREGATION_BATCH_SIZE)
+                .sources(sources);
+
+        if (afterKey != null) {
+            compositeAggregationBuilder.after(afterKey);
+        }
+
+        return compositeAggregationBuilder.build();
+    }
+
+    private NativeQuery buildNativeQuery(String compositeName, CompositeAggregation compositeAggregation) {
+        Aggregation aggregation = Aggregation.of(a -> a.composite(compositeAggregation));
+
+        return new NativeQueryBuilder()
+                .withAggregation(compositeName, aggregation)
+                .build();
+    }
+
+    private Pair<List<Map<String, FieldValue>>, Map<String, FieldValue>> processSearchResults(SearchHits<EquipmentInfos> searchHits, String compositeName) {
+        ElasticsearchAggregations aggregations = (ElasticsearchAggregations) searchHits.getAggregations();
+
+        List<Map<String, FieldValue>> results = new ArrayList<>();
+        if (aggregations != null) {
+            Map<String, ElasticsearchAggregation> aggregationList = aggregations.aggregationsAsMap();
+            if (!aggregationList.isEmpty()) {
+                Aggregate aggregate = aggregationList.get(compositeName).aggregation().getAggregate();
+                if (aggregate.isComposite() && aggregate.composite() != null) {
+                    for (CompositeBucket bucket : aggregate.composite().buckets().array()) {
+                        Map<String, FieldValue> key = bucket.key();
+                        results.add(key);
+                    }
+                    return Pair.of(results, aggregate.composite().afterKey());
+                }
+            }
+        }
+        return Pair.of(results, null);
+    }
+
+    public List<UUID> getEquipmentInfosDistinctNetworkUuids() {
+        List<UUID> networkUuids = new ArrayList<>();
```

Comment:
```suggestion
        List<UUID> networkUuids = new LinkedList<>();
```
As we don't know the size in advance, and there will be a lot of elements in standard cases, a more optimized implementation of `List` is preferable.